internal/errorhandling/safe.go

```diff
@@ -0,0 +1,36 @@
+package errorhandling
+
+import "fmt"
+
+// safe2 runs the specified function and returns its result value or returned error. If a panic occurs, it returns the
+// panic as an error.
+//
+// Note: this is equivalent to a try-catch and you should probably not use it. Only use if you need to handle
+// panics from third party libraries or from Golang itself.
+func safe2[TValue any](f func() (TValue, error)) (result TValue, err error) {
+	defer func() {
+		var ok bool
+		e := recover()
+		if e == nil {
+			return
+		}
+		if err, ok = e.(error); !ok {
+			// In case the panic is not an error
+			err = fmt.Errorf("%v", e)
+		}
+	}()
+	return f()
+}
+
+// Safe2 runs the specified function and returns its result value or returned error. If a panic occurs, it returns the
+// panic as an error. Any errors returned are passed through the passed wrap function.
+//
+// Note: this is equivalent to a try-catch and you should probably not use it. Only use if you need to handle
+// panics from third party libraries or from Golang itself.
+func Safe2[TValue any](f func() (TValue, error), wrap func(err error) error) (result TValue, err error) {
```

Comment:
My main issue is that Go doesn't have method / function overrides, which would be really helpful here. Some functions may not have a return value, which would then need a function called `Safe1`. On the flip side, some functions may return two values, or some function may not return an error but only panic. The naming should be such that it's extensible to that end. (I have removed these functions at the request of the review in the engineering meeting because we don't need them now.)

Reply:
Regarding the callsites: we can, but it makes the code a bit harder to read.

Reply:
Is there a reason to make this it's own function at this point?  Should we wait until there are multiple instances to refactor it out?

Reply:
I tried implementing it inline and it became very disgusting. I could make this a package-internal function, but that feels wrong to me because it has no business being in the `aesgcm` package. We should also encourage code reuse rather than discourage it.